float64 d_left
float64 d_right